src/Helper/AutomatedEditsHelper.php

```diff
@@ -37,16 +37,17 @@ public function __construct(RequestStack $requestStack, CacheItemPoolInterface $
     }
 
     /**
-     * Get the tool that matched the given edit summary.
-     * This only works for tools defined with regular expressions, not tags.
+     * Get the first tool that matched the given edit summary and tags.
      * @param string $summary Edit summary
      * @param Project $project
+     * @param string[]|null $tags
      * @return string[]|null Tool entry including key for 'name', or false if nothing was found
      */
-    public function getTool(string $summary, Project $project): ?array
+    public function getTool(string $summary, Project $project, ?array $tags = []): ?array
     {
         foreach ($this->getTools($project) as $tool => $values) {
-            if (isset($values['regex']) && preg_match('/'.$values['regex'].'/', $summary)) {
+            if ((isset($values['regex']) && preg_match('/'.$values['regex'].'/', $summary)) ||
+                (isset($values['tags']) && count(array_intersect($values['tags'], $tags)) > 0)) {
```

Comment:
What happens if `$tags` is `null`? It looks like this would error out. Does it need to be nullable?

Nitpick:
```suggestion
                (isset($values['tags']) && count(array_intersect($values['tags'], $tags)) > 0))
            {
```

Reply:
TopEdits also uses this, though isAutomated(); and TopEdits doesn't include the tags. However, it's true that the default value of [] is not null, and so it doesn't need to be nullable.